requirements-dev.txt

```diff
@@ -1 +1 @@
--e .[dev]
+-e .[dev,docs]
```

Comment:
This seems a little odd - shouldn't requirements-dev match `[dev]`?  and `[dev]` include the requirements for building docs? Could the `[dev]` extra depend on other extras?

Reply:
Extras can't depend on other extras (my biggest pet peeve - and one that Poetry seems to not have solved when they built a more complicated requirements system), so they'd have to be listed twice. However, this means there might something missing from docs. I thought it was a little nicer to split "docs" from "dev", as often developers are not rebuilding the docs, so it gives more freedom there (since it's easy to ask for multiple extras. You can't as easily make multiple requirements files (it's ugly), and for historical purposes, the requirements-dev should stay the same. But I can also list them twice if better.

Reply:
Note: you _can_ move extras back to setup.py, then you can make them depend on either other via code, I do this for extras sometimes: https://github.com/scikit-hep/boost-histogram/blob/918caebdf922097fde92df5a5de5442d86aa04c7/setup.py#L51-L64

I often have an "all" if I do this this way.